Formula/g/gitsign.rb

```diff
@@ -0,0 +1,27 @@
+class Gitsign < Formula
+  desc "Keyless Git signing using Sigstore"
+  homepage "https://github.com/sigstore/gitsign"
+  url "https://github.com/sigstore/gitsign/archive/refs/tags/v0.7.1.tar.gz"
+  sha256 "b9b82840db5181e44e9979b464844c34dba4cb1f8cfedbc3650d5347f0dcf5c6"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/sigstore/gitsign/pkg/version.gitVersion=#{version}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags)
+
+    generate_completions_from_executable(bin/"gitsign", "completion")
+  end
+
+  test do
+    output = shell_output("#{bin}/gitsign --version")
+
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
In this case the binary prints it's configuration details when running `gitsign --version`. So we're checking that it's using the correct issuer. Additional functional tests would require cloning a repo and verifying a commit, but I'm not sure we want to do something like that.

Reply:
Actually I saw that other formulae use `git clone` as part of their tests, so I wrote a new one that uses that.